hibernate-core/src/main/java/org/hibernate/cfg/ToOneFkSecondPass.java

```diff
@@ -67,29 +67,19 @@ else if ( property != null) {
 			return path.startsWith( property.getName() + "." );
 		}
 		//try the embedded property
-		//embedded property starts their path with 'id.' See PropertyPreloadedData( ) use when idClass != null in AnnotationSourceProcessor
-		else if ( path.startsWith( "id." ) ) {
-			KeyValue valueIdentifier = persistentClass.getIdentifier();
-			String localPath = path.substring( 3 );
+		else {
+			final KeyValue valueIdentifier = persistentClass.getIdentifier();
 			if ( valueIdentifier instanceof Component ) {
-				Iterator it = ( (Component) valueIdentifier ).getPropertyIterator();
-				while ( it.hasNext() ) {
-					Property idProperty = (Property) it.next();
-					if ( localPath.startsWith( idProperty.getName() ) ) {
-						return true;
-					}
+				// Embedded property starts their path with 'id.'
+				// See PropertyPreloadedData( ) use when idClass != null in AnnotationSourceProcessor
+				String localPath = path;
+				if ( path.startsWith( "id." ) ) {
+					localPath = path.substring( 3 );
 				}
-			}
-		}
-		// Try the case where a @ManyToOne is also an ID property
-		// E.g. @ManyToOne @Id SomeEntity other;
-		else if ( !path.contains( "." ) ) {
-			KeyValue valueIdentifier = persistentClass.getIdentifier();
-			if ( valueIdentifier instanceof Component ) {
 				Iterator it = ( (Component) valueIdentifier ).getPropertyIterator();
 				while ( it.hasNext() ) {
 					Property idProperty = (Property) it.next();
-					if ( path.equals( idProperty.getName() ) ) {
+					if ( localPath.startsWith( idProperty.getName() ) ) {
```

Comment:
My main concern with this `startsWith` business, and the reason I used `equals`, is that a path can start with the name of an ID property without being related to it.

For example, IIUC, the following model would trigger a false positive for the association `EmployeeRecord.employeeStatus`: it's not in the ID, but it would be detected as such because it starts with `employee`, which is an ID property.

```
@Entity
public class EmployeeRecord {

    @Id
    @ManyToOne
    Company company;

    @Id
    @ManyToOne
    Employee employee;

    @ManyToOne
    EmployeeStatus employeeStatus;

}
```

Maybe we should go with something like this instead?

```suggestion
					if ( localPath.equals( idProperty.getName() ) || localPath.startsWith( idProperty.getName() + "." ) ) { 
```